Sources/StreamVideoSwiftUI/CallView/LayoutComponents/BottomParticipantsBarLayoutComponent.swift

```diff
@@ -0,0 +1,114 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import StreamVideo
+import SwiftUI
+
+/// `BottomParticipantsBarLayoutComponent` represents a horizontally scrollable view of participant thumbnails.
+/// This component lays out participant thumbnails in a bar at the bottom of the associated view.
+public struct BottomParticipantsBarLayoutComponent<Factory: ViewFactory>: View {
+
+    // MARK: - Properties
+
+    /// Factory for creating views.
+    public var viewFactory: Factory
+
+    /// List of participants to display.
+    public var participants: [CallParticipant]
+
+    /// Frame in which the component will be laid out.
+    public var frame: CGRect
+
+    /// Information about the call (if available).
+    public var call: Call?
+
+    /// Size of each participant thumbnail.
+    public var thumbnailSize: CGFloat
+
+    /// Flag to determine if all participant information should be shown.
+    public var showAllInfo: Bool
+
+    /// Closure called to change visibility of a participant's track.
+    public var onChangeTrackVisibility: @MainActor(CallParticipant, Bool) -> Void
+
+    // Private computed properties for laying out the view.
+    private let barFrame: CGRect
+    private let itemFrame: CGRect
+
+    // MARK: - Initialization
+
+    /// Creates a new instance of `BottomParticipantsBarLayoutComponent`.
+    public init(
+        viewFactory: Factory,
+        participants: [CallParticipant],
+        frame: CGRect,
+        call: Call?,
+        thumbnailSize: CGFloat = 120,
+        showAllInfo: Bool = false,
+        onChangeTrackVisibility: @escaping (CallParticipant, Bool) -> Void
+    ) {
+        self.viewFactory = viewFactory
+        self.participants = participants
+        self.frame = frame
+        self.call = call
+        self.thumbnailSize = thumbnailSize
+        self.showAllInfo = showAllInfo
+        self.onChangeTrackVisibility = onChangeTrackVisibility
+
+        // Calculate the frame for the bar at the bottom.
+        self.barFrame = .init(
+            origin: .init(x: frame.origin.x, y: frame.maxY - thumbnailSize),
+            size: CGSize(width: frame.size.width, height: thumbnailSize)
+        )
+
+        // Calculate the frame for each item in the bar.
+        self.itemFrame = .init(
```

Comment:
How's this different from the thumbnail size?

Reply:
It isn't in reality but it's easier to use. The thumbnailSize refer to the size of one dimension (that we then use for both width and height). We create this frame so we can pass it in the creation of each thumbnail without having to go through this `CGRect(origin: .zero, size: .init(width: thumbnailSize, height: thumbnailSize))`